[CI] TermsAggregatorTests.testManyUniqueTerms failure

**Build scan**:
https://gradle-enterprise.elastic.co/s/zs7wpexyuupcg

**Repro line**:
```
./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.terms.TermsAggregatorTests.testManyUniqueTerms" \
  -Dtests.seed=928BECEE0B850D70 \
  -Dtests.security.manager=true \
  -Dtests.locale=ar-SD \
  -Dtests.timezone=America/Indiana/Knox \
  -Druntime.java=8
```

**Reproduces locally?**:
YES

**Applicable branches**:
7.12

**Failure history**:
https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-90d,mode:quick,to:now))&_a=(columns:!(_source),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:testManyUniqueTerms),sort:!(time,desc))

**Failure excerpt**:
```
10:46:22   2> java.lang.AssertionError: 
10:46:22     Expected: <[b007, b107, b207, b307, b407, b507, b607, b707, b000, b001]>
10:46:22          but: was <[b007, b107, b207, b307, b407, b507, b707, b000, b001, b002]>
10:46:22         at __randomizedtesting.SeedInfo.seed([928BECEE0B850D70:E65ADE16B0EDEB46]:0)
10:46:22         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
10:46:22         at org.junit.Assert.assertThat(Assert.java:956)
10:46:22         at org.junit.Assert.assertThat(Assert.java:923)
10:46:22         at org.elasticsearch.search.aggregations.bucket.terms.TermsAggregatorTests.lambda$testManyUniqueTerms$6(TermsAggregatorTests.java:304)
10:46:22         at org.elasticsearch.search.aggregations.AggregatorTestCase.testCase(AggregatorTestCase.java:545)
10:46:22         at org.elasticsearch.search.aggregations.bucket.terms.TermsAggregatorTests.testManyUniqueTerms(TermsAggregatorTests.java:295)
```

--- a/server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorTests.java
+++ b/server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorTests.java
@@ -286,19 +286,27 @@ public class TermsAggregatorTests extends AggregatorTestCase {
         }
     }
 
-    public void testManyUniqueTerms() throws Exception {
+    public void testManyTerms() throws Exception {
         MappedFieldType fieldType = new KeywordFieldMapper.KeywordFieldType("string", randomBoolean(), true, null);
         TermsAggregationBuilder aggregationBuilder = new TermsAggregationBuilder("_name")
             .executionHint(randomFrom(TermsAggregatorFactory.ExecutionMode.values()).toString())
             .field("string");
         testCase(aggregationBuilder, new MatchAllDocsQuery(), iw -> {
+            /*
+             * index all of the fields into a single segment so our
+             * test gets accurate counts. We *could* set the shard size
+             * very very high but we want to test the branch of the
+             * aggregation building code that picks the top sorted aggs.
+             */
+            List<List<? extends IndexableField>> docs = new ArrayList<>();
             for (int i = 0; i < TermsAggregatorFactory.MAX_ORDS_TO_TRY_FILTERS - 200; i++) {
                 String s = String.format(Locale.ROOT, "b%03d", i);
-                iw.addDocument(doc(fieldType, s));
+                docs.add(doc(fieldType, s));
                 if (i % 100 == 7) {
-                    iw.addDocument(doc(fieldType, s));
+                    docs.add(doc(fieldType, s));
                 }
             }
+            iw.addDocuments(docs);
         }, (StringTerms result) -> {
             assertThat(result.getBuckets().stream().map(StringTerms.Bucket::getKey).collect(toList()),
                 equalTo(List.of("b007", "b107", "b207", "b307", "b407", "b507", "b607", "b707", "b000", "b001")));

